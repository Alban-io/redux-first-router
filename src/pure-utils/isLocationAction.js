// @flow
import type { Action, ReceivedAction } from '../flow-types'


export default (action: Action | ReceivedAction): boolean =>
  !!(action.meta && action.meta.location)
