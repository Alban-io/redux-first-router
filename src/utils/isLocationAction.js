// @flow
import type { Action, ReceivedAction } from '../flow-types'

export default (action: Object): boolean =>
  action.location && action.location.url
