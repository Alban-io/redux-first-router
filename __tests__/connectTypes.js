import isLocationAction from '../src/pure-utils/isLocationAction'


describe('isLocationAction', () => {
  it('isLocationAction snap', () => {
    const ret = {d: 3}
    expect(ret).toMatchSnapshot()
  })

  it('isLocationAction 123', () => {
    const ret = {d: 8}
    expect(ret).toMatchSnapshot()
  })
})
