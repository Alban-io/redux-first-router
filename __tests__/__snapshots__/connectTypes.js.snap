exports[`enhancer -> _historyAttemptDispatchAction() dispatches action matching pathname when history location changes 1`] = `
Object {
  "meta": Object {
    "location": Object {
      "backNext": true,
      "current": Object {
        "pathname": "/second/foo",
        "payload": Object {
          "param": "foo",
        },
        "type": "SECOND",
      },
      "load": undefined,
      "prev": Object {
        "pathname": "",
        "payload": Object {},
        "type": "",
      },
    },
  },
  "payload": Object {
    "param": "foo",
  },
  "type": "SECOND",
}
`;

exports[`enhancer dispatches location-aware action when store is first created so app is location aware on load 1`] = `
Object {
  "meta": Object {
    "location": Object {
      "backNext": undefined,
      "current": Object {
        "pathname": "/first",
        "payload": Object {},
        "type": "FIRST",
      },
      "load": true,
      "prev": Object {
        "pathname": "",
        "payload": Object {},
        "type": "",
      },
    },
  },
  "payload": Object {},
  "type": "FIRST",
}
`;

exports[`enhancer listens to history changes and dispatches actions matching history's location.pathname 1`] = `
Object {
  "meta": Object {
    "location": Object {
      "backNext": true,
      "current": Object {
        "pathname": "/second/bar",
        "payload": Object {
          "param": "bar",
        },
        "type": "SECOND",
      },
      "load": undefined,
      "prev": Object {
        "pathname": "/first",
        "payload": Object {},
        "type": "FIRST",
      },
    },
  },
  "payload": Object {
    "param": "bar",
  },
  "type": "SECOND",
}
`;

exports[`middleware dispatches location-aware action, changes address bar + document.title 1`] = `
Object {
  "meta": Object {
    "location": Object {
      "backNext": undefined,
      "current": Object {
        "pathname": "/second/bar",
        "payload": Object {
          "param": "bar",
        },
        "type": "SECOND",
      },
      "load": undefined,
      "prev": Object {
        "pathname": "",
        "payload": Object {},
        "type": "",
      },
    },
  },
  "payload": Object {
    "param": "bar",
  },
  "type": "SECOND",
}
`;

exports[`middleware dispatches location-aware action, changes address bar + document.title 2`] = `
Object {
  "location": Object {
    "backNext": undefined,
    "load": undefined,
    "pathname": "/second/bar",
    "payload": Object {
      "param": "bar",
    },
    "prev": Object {
      "pathname": "",
      "payload": Object {},
      "type": "",
    },
    "routesMap": Object {
      "FIRST": "/first",
      "SECOND": "/second/:param",
    },
    "type": "SECOND",
  },
  "title": "title: SECOND",
}
`;

exports[`middleware user dispatches NOT_FOUND and middleware adds missing info to action 1`] = `
Object {
  "meta": Object {
    "location": Object {
      "backNext": undefined,
      "current": Object {
        "pathname": "/",
        "payload": Object {},
        "type": "@@pure-redux-router/NOT_FOUND",
      },
      "load": undefined,
      "prev": Object {
        "pathname": "",
        "payload": Object {},
        "type": "",
      },
    },
  },
  "payload": Object {},
  "type": "@@pure-redux-router/NOT_FOUND",
}
`;

exports[`test reducer EXISTS and works (see __tests__/createLocationReducer for all its tests) 1`] = `
Object {
  "backNext": true,
  "load": true,
  "pathname": "/second/bar",
  "payload": Object {
    "param": "bar",
  },
  "prev": Object {
    "pathname": "/first",
    "payload": Object {},
    "type": "FIRST",
  },
  "routesMap": Object {
    "FIRST": "/first",
    "SECOND": "/second/:param",
  },
  "type": "SECOND",
}
`;
